src/format.rs

```diff
@@ -3,47 +3,160 @@ use crate::__gl;
 #[allow(non_camel_case_types)]
 #[repr(u32)]
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+/// The `Format` enum represents a sized internal format of texture
+/// storage. The naming convention is similar to that of the internal
+/// format OpenGL constants, with some extensions for increased
+/// consistency:
+///
+/// * The '_UNORM' suffix denotes unsigned normalized formats. They are
+/// represented as unsigned integers internally, remapped to the [0.0,
+/// 1.0] floating point range in shaders. These are equivalent to the
+/// constants with no suffix in OpenGL.
+///
+/// * The '_SNORM' suffix denotes signed normalized formats. They are
+/// represented as signed integers, remapped to the [-1.0, 1.0]
+/// floating point range in shaders. This suffix is the same as in
+/// OpenGL.
+///
+/// * The '_SFLOAT' suffix denotes floating point formats, equivalent
+/// to the 'F' suffix.
+///
+/// * The '_I' suffix denotes signed integer formats, exposed to
+/// shaders unmodified as integers.
+///
+/// * The '_UI' suffix denotes unsigned integer formats, exposed to
+/// shaders as unsigned integers.
+///
+/// * The '_SRGB' suffix denotes sRGB formats, which are all unsigend
+/// normalized integers. Textures in this format are assumed to be in
+/// the sRGB colorspace. Shaders reading from this format will
+/// automatically convert the color components to a linear colorspace,
+/// so the shader will only see linear values. When outputting to a
+/// render target with an '_SRGB' format, OpenGL will convert the
+/// color components to an sRGB colorspace iff the `GL_FRAMEBUFFER_SRGB` is
+/// enabled. Alpha components, if they exist, are treated as linear
+/// throughout.
+///
+/// Each component is followed by the number of bits used to represent
+/// it.
 pub enum Format {
+    // unsigned normalized integer formats
     R8_UNORM = __gl::R8,
-    R8G8B8A8_SRGB = __gl::RGBA8,
-    R8G8B8A8_UNORM = __gl::SRGB8_ALPHA8,
+    R8G8_UNORM = __gl::RG8,
+    R8G8B8_UNORM = __gl::RGB8,
+    R8G8B8A8_UNORM = __gl::RGBA8,
```

Comment:
I did change how `R8G8B8A8_SRGB` and `R8G8B8A8_UNORM` are defined, as they didn't seem to consistent.